airflow-core/src/airflow/api_fastapi/core_api/openapi/v2-rest-api-generated.yaml

```
@@ -2135,6 +2135,87 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/HTTPValidationError'
+  /api/v2/dags/{dag_id}/dagRuns/{dag_run_id}/wait:
```

```suggestion
  /api/v2/dags/{dag_id}/dagRuns/{dag_run_id}/watch:
```

Nit: To me it looks like we're watching the dag run state, and stop watching when it's settled. But I'm good with wait too.